cpp/src/io/parquet/page_data.cu

```diff
@@ -759,6 +759,24 @@ inline __device__ void gpuOutputInt64Timestamp(volatile page_state_s* s, int src
   *dst = ts;
 }
 
+/**
+ * @brief Output a byte array as int.
+ *
+ * @param[in] ptr Pointer to the byte array
+ * @param[in] len Byte array length
+ * @param[out] dst Pointer to row output data
+ */
+template <typename T>
+__device__ void gpuOutputByteArrayAsInt(char const* ptr, int32_t len, T* dst)
```

Comment:
It looks like negative values are not being handled properly with this.

`0xFF` by itself should result in a `-1` being returned. But we are treating it the same as `0x00, 0xFF`. The sign of the Most Significant Byte should be extended through the rest of `dst`.

Reply:
Fixed